src/js/dp/map.js

```diff
@@ -0,0 +1,233 @@
+};
+/** @internal */
+const _getMapCenter = (mapEl) => {
+    let mapCenter = mapEl.dataset.center;
+    if (!mapCenter) {
+        throw new Error("You must declare a 'data-center' attribute");
+    }
+    mapCenter = mapCenter.replace(/\[|]/g, "");
+    if (mapCenter.length < 2) throw new Error(centerErrorMsg);
+    mapCenter = mapCenter.split(",");
+    mapCenter = mapCenter.map(center => {
+        return parseInt(center);
+    });
+    return mapCenter;
+};
+/** @internal */
+const _getMapZoom = (mapEl) => {
+    let mapZoom = parseInt(mapEl.dataset.zoom)
+    if (!mapZoom) {
+        mapZoom = 3;
+        console.warn("OSD Map Component: defaulting the zoom value to 3");
+    } else {
+        mapZoom =  parseInt(mapEl.dataset.zoom)
+    }
+    return mapZoom;
+};
+/** @internal */
+const _getToken = (mapEl) => {
+    return mapEl.dataset.token;
+}
+/** @internal */
+class MapState {
+    constructor() {
+        this.context = {
+            hasMoved: false,
+            canvasTabbed: false,
+        };
+    }
+    
+    update(state = {}) {
+        this.context = { 
+            ...this.context,
+             ...state,
+        };
+    }
+}
+/** @internal */
+const Keys = Object.freeze({
+    R: 82,
+    ESCAPE: 9,
+});
+/** @internal */
+class ONSCustomControl {
+    _controlElements = [];
+
+    constructor(center, zoom, mapState) {
+        this.mapState = mapState;
+        this.center = center;
+        this.zoom = zoom;
+    }
+
+    set controlElements(val) {
+        if (!this._controlElements.length) {
+            this._controlElements = val.map((elem, i) => {
+                elem.setAttribute("tabindex", (0).toString());
+                return elem;
+            });
+        } else {
+            throw new Error("Attribute 'controlElements' already set");
+        }
+    }
+
+    get controlElements() {
+        return this._controlElements;
+    }
+
+    onAdd(map) {
+        this.map = map;
+        this.container = document.createElement("button");
+        this.container.className = "mapboxgl-ctrl-reset";
+        this.container.addEventListener("click", this._onReset);
+        this.map.getCanvas().addEventListener("keydown", this._onKeyDown);
+        return this.container;
+    }
+
+    onRemove() {
+        this.container.parentNode.removeChild(this.container);
+        this.map = undefined;
+    }
+
+    _onReset = (e) => {
+        this.map.flyTo({
+            center: this.center,
+            zoom: this.zoom,
+        });
+        this.container.firstChild.className = "mapboxgl-ctrl-icon mapboxgl-ctrl-icon--disabled";
+        this.mapState.update({ hasMoved: false });
+        this.container.disabled = true;
+    };
+
+    _onKeyDown = (e) => {
+        e.preventDefault();
+        const canvas = this.map.getCanvas();
+        if (e.which === Keys.R) {
+            this._onReset(e);
+            return;
+        }
+        if (e.which === Keys.ESCAPE) {
+            canvas.blur();
+        }
+
```

Comment:
```suggestion
switch (e.which) {
    case Keys.R:
        this._onReset(e);
        break;
    case Keys.Escape:
        canvas.blur(); 
        break;
}
```

or `if` ... `else if` without the `return` in the first block if `switch` isn't the house style